script.module.clouddrive.common/clouddrive/common/export.py

```diff
@@ -0,0 +1,160 @@
+                        with open(config_path, 'rb') as fo:
+                            items_info = json.loads(fo.read())
+                            self.export_items_db.set(exportid, items_info)
+                        os.rename(config_path, config_path + '.migrated')
+                except Exception as ex:
+                    Logger.debug("Error migrating export items from %s" % filename)
+                    Logger.debug(ex)
+                    os.rename(config_path, config_path + '.failed')
+                
+        
+    def get_exports(self):
+        return self.exports_db.getall()
+
+    @timeit
+    def save_export(self, export):
+        self.exports_db.set(export['id'], export)
+
+    def remove_export(self, exportid, keep_local=True):
+        export = self.exports_db.get(exportid)
+        self.exports_db.remove(exportid)
+        self.export_items_db.remove(exportid)
+        self.export_items_db.remove('pending-' + exportid)
+        self.export_items_db.remove('retry-' + exportid)
+        path = os.path.join(export['destination_folder'], Utils.unicode(export['name']), '')
+        if not keep_local:
+            if KodiUtils.file_exists(path):
+                Utils.remove_folder(path)
+
+    def get_items_info(self, exportid):
+        return self.export_items_db.get(exportid)
+
+    @timeit
+    def save_items_info(self, exportid, items_info):
+        self.export_items_db.set(exportid, items_info)
+    
+    def get_pending_changes(self, exportid):
+        return deque(Utils.default(self.export_items_db.get('pending-' + exportid), []))
+
+    @timeit
+    def save_pending_changes(self, exportid, changes):
+        self.export_items_db.set('pending-' + exportid, list(changes))
+
+    def get_retry_changes(self, exportid):
+        return deque(Utils.default(self.export_items_db.get('retry-' + exportid), []))
+
+    @timeit
+    def save_retry_changes(self, exportid, changes):
+        self.export_items_db.set('retry-' + exportid, list(changes))
+                
+    @staticmethod
+    def add_item_info(items_info, item_id, name, full_local_path, parent, item_type):
+        items_info[item_id] = {'name': name, 'full_local_path': full_local_path, 'parent': parent,'type':item_type}
+
+    @staticmethod
+    def remove_item_info(items_info, item_id):
+        if item_id in items_info:
+            del items_info[item_id]
+
+    @staticmethod
+    def get_strm_link(driveid, item, content_type, addon_url):
+        item_id = Utils.str(item['id'])
+        item_drive_id = Utils.default(Utils.get_safe_value(item, 'drive_id'), driveid)
+        content = addon_url + '?' + urllib.parse.urlencode(
+                {'action': 'play', 'content_type': content_type, 'item_driveid': item_drive_id, 'item_id': item_id,
+                 'driveid': driveid})
+        return Utils.str(content)
+    
+    @staticmethod
+    def create_text_file(file_path, content):
+        f = None
+        try:
```

Comment:
if this is xbmcvfs, kodi supports contextlib for these calls since matrix

Reply:
changed